osbenchmark/workload_generator/workload_generator.py

```diff
@@ -61,6 +62,14 @@ def extract_mappings_and_corpora(client, output_path, indices_to_extract):
 
     return indices, corpora
 
+def process_custom_queries(custom_queries):
+    if not custom_queries:
+        return []
+
+    with custom_queries as queries:
+        data = json.load(queries)
```

Comment:
It would be good to have some error reporting here, indicating the user's queries is not valid JSON, rather than relying upon the `json` module's error messages.